src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitedArray = date.split(fromFormat[fromFormat.length - 1]);
+  let year = '';
+  const separator = toFormat[toFormat.length - 1];
+
+  toFormat.pop();
+
+  for (const i of toFormat) {
+    if (i === 'DD') {
+      toFormat[toFormat.indexOf(i)] = splitedArray[fromFormat.indexOf(i)];
+    }
+
+    if (i === 'MM') {
+      toFormat[toFormat.indexOf(i)] = splitedArray[fromFormat.indexOf(i)];
+    }
+
+    if ((i === 'YYYY' || i === 'YY') && fromFormat.includes(i)) {
+      year = splitedArray[fromFormat.indexOf(i)];
+      toFormat[toFormat.indexOf(i)] = year;
```

Comment:
The condition `fromFormat.includes(i)` is redundant because the `toFormat` array is based on the `fromFormat` array. If 'YYYY' or 'YY' is present in `toFormat`, it must have come from `fromFormat`.